### PR TITLE
Local keyCode to Keycode-js switch

### PR DESCRIPTION
**Pull request by AlexBrizi**
### Summary
Replaced all instances of local keyCode values with the [keycode-js npm package](https://www.npmjs.com/package/keycode-js).

Resolves [#2235](https://github.com/cerner/terra-core/issues/2235)

### CODE REVIEW

**Comment by AlexBrizi** [comment]
Not sure why my Pull Requests keep getting merged with previous PRs... I ensured I was on the correct branch before pushing. Any advice/suggestions would be appreciated

File: packages/terra-form-select/src/_Frame.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
-import { KeyCodes, Variants } from './_constants';
+import { KeyCode, Variants } from './_constants';
```
Comment (emilyrohrbough) [inline packages/terra-form-select/src/_Frame.jsx]: I feel like we should directly import from the keycode package and remove the keycode exports from `_constants.js`

File: packages/terra-form-select/src/_Menu.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import { polyfill } from 'react-lifecycles-compat';
 import 'terra-base/lib/baseStyles';
-import { KeyCodes, Variants } from './_constants';
+import { KeyCode, Variants } from './_constants';
```
Comment (emilyrohrbough) [inline packages/terra-form-select/src/_Menu.jsx]: I feel like we should directly import from the keycode package and remove the keycode exports from `_constants.js`

File: packages/terra-paginator/src/ProgressivePaginator.jsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import ResponsiveElement from 'terra-responsive-element';
 import 'terra-base/lib/baseStyles';
 import styles from './Paginator.module.scss';
 
-import { calculatePages, KEYCODES } from './_paginationUtils';
+import { calculatePages, KeyCode } from './_paginationUtils';
```
Comment (emilyrohrbough) [inline packages/terra-paginator/src/ProgressivePaginator.jsx]: I feel like we should directly import from the keycode package and remove the keycode exports from `_paginationUtils.js`

File: packages/terra-markdown/package.json [hunk 4]
```diff
@@ -36,7 +37,8 @@
     "lint": "npm run lint:js && npm run lint:scss",
     "lint:js": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
     "lint:scss": "stylelint src/**/*.scss",
-    "test": "npm run test:jest",
-    "test:jest": "jest --config ../../jestconfig.js"
+    "test": "npm run test:jest && npm run test:wdio",
+    "test:jest": "jest --config ../../jestconfig.js",
+    "test:wdio": "../../node_modules/.bin/wdio ../../node_modules/terra-dev-site/config/wdio/wdio.conf.js"
```
Comment (emilyrohrbough) [inline packages/terra-markdown/package.json]: Looks like there are some non-keycode changes in the pr

**Comment by emilyrohrbough** [comment]
@AlexBrizi How are you creating a new branch?

I always match sure I am on the master branch before creating a new branch so the new brach has the latest master changes and commits, not the branch changes of the previous branch
```bash
> git checkout master
> git pull
> git checkout -b new_branch
```

**Comment by yuderekyu** [comment]
@Alex-Brizi A new branch will start from the branch you're currently on. In this case, you branched off of the `color-blockquotes` branch. The commands @emilyrohrbough posted will make sure you start fresh from `master` 👍

File: packages/terra-button-group/CHANGELOG.md [hunk 5]
```diff
@@ -3,10 +3,13 @@ Changelog
 
 Unreleased
 ----------
+### Changed
+* Replaced local keyCode values with keycode-js npm package 
 
 3.6.0 - (February 26, 2019)
 ------------------
 ### Changed
+* Added a11y clarity to not-selectable button group
```
Comment (bjankord) [inline packages/terra-button-group/CHANGELOG.md]: You can remove this line, it is covered by the bullet point below
```
* Improved screen reader clarity on non-selectable button group buttons
```

File: packages/terra-markdown/CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
```
Comment (bjankord) [inline packages/terra-markdown/CHANGELOG.md]: This addition can be removed, it is already in this file on lines 14 and 15

File: packages/terra-paginator/src/ControlledPaginator.jsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ import ResponsiveElement from 'terra-responsive-element';
 import 'terra-base/lib/baseStyles';
 import styles from './Paginator.module.scss';
 
-import { calculatePages, pageSet, KEYCODES } from './_paginationUtils';
+import { calculatePages, pageSet, KeyCode } from './_paginationUtils';
```
Comment (bjankord) [inline packages/terra-paginator/src/ControlledPaginator.jsx]: Let's update this to pull KeyCode from keycode-js directly instead.

```
import KeyCode from 'keycode-js';
import { calculatePages, pageSet } from './_paginationUtils';
```

File: packages/terra-paginator/src/ControlledProgressivePaginator.jsx [hunk 8]
```diff
@@ -6,7 +6,7 @@ import ResponsiveElement from 'terra-responsive-element';
 import 'terra-base/lib/baseStyles';
 import styles from './Paginator.module.scss';
 
-import { calculatePages, KEYCODES } from './_paginationUtils';
+import { calculatePages, KeyCode } from './_paginationUtils';
```
Comment (bjankord) [inline packages/terra-paginator/src/ControlledProgressivePaginator.jsx]: Let's update this to pull KeyCode from keycode-js directly instead.

```
import KeyCode from 'keycode-js';
import { calculatePages } from './_paginationUtils';
```

File: packages/terra-paginator/src/Paginator.jsx [hunk 9]
```diff
@@ -6,7 +6,7 @@ import ResponsiveElement from 'terra-responsive-element';
 import 'terra-base/lib/baseStyles';
 import styles from './Paginator.module.scss';
 
-import { calculatePages, pageSet, KEYCODES } from './_paginationUtils';
+import { calculatePages, pageSet, KeyCode } from './_paginationUtils';
```
Comment (bjankord) [inline packages/terra-paginator/src/Paginator.jsx]: Let's update this to pull KeyCode from keycode-js directly instead.
```
import KeyCode from 'keycode-js';
import { calculatePages, pageSet } from './_paginationUtils';
```

File: packages/terra-paginator/src/_paginationUtils.jsx [hunk 10]
```diff
@@ -1,3 +1,5 @@
+import KeyCode from 'keycode-js';
```
Comment (bjankord) [inline packages/terra-paginator/src/_paginationUtils.jsx]: Let's drop the KeyCode import and export from this file and instead import it directly where needed.

**Comment by bjankord** [comment]
These code changes look good to me. However, the image diffs are unexpected. I've merged master into this branch 

@AlexBrizi, when you start work on this branch again I'd recommend pulling down the latest changes.
Run `git pull` in your `keycode-js` branch.
Then you'll want to check out the master branch and pull down the latest changes for that branch.

```
git checkout master
git pull
```
Then you'll want to switch back to your feature branch
```
git checkout keycode-js
```
The next step will be to revert the wdio images to the images that are in master.
Run the following commands individualy 
```
git checkout origin/master packages/terra-table/tests/wdio/
git checkout origin/master packages/terra-search-field/tests/wdio/
git checkout origin/master packages/terra-paginator/tests/wdio/
git checkout origin/master packages/terra-overlay/tests/wdio/
git checkout origin/master packages/terra-markdown/tests/wdio/
git checkout origin/master packages/terra-hyperlink/tests/wdio/
git checkout origin/master packages/terra-form-select/tests/wdio/
git checkout origin/master packages/terra-button/tests/wdio/
git checkout origin/master packages/terra-button-group/tests/wdio/
```
This should reset the images in those packages back to what is the equvialent in the master branch which would mean no image diffs on this PR. I'm curious if you run the wdio tests for those packages after that if you see any image diffs. It would be expected that there are no image diffs when you run those tests.

**Pull request merged**